### PR TITLE
Add the improvements from :class:`~.VGroup` to :class:`~.Group`

### PR DESCRIPTION
**Pull request by WampyCakes**
## Explanation for Changes
Adds the improvements from VGroup to Group and makes them more consistent.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by WampyCakes** [comment]
I think this would be reasonable to get into v0.5.0 since it's really nothing new.

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -2128,12 +2128,75 @@ def set_z_index_by_z_coordinate(self):
...
+    in order to scale, move, or perform other animations on them together.
+
+    To add :class:`~.Mobject`s to a :class:`~.Group`, you can either use the
+    :meth:`~.Group.add` method, or use the `+` and `+=` operators. Similarly, you
+    can subtract elements of a Group via :meth:`~.Group.remove` method, or
+    `-` and `-=` operators:
+
+    See Also
+    --------
+    :class:`VGroup`
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:2143]:
```suggestion
    :class:`~.VGroup`
```

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -2128,12 +2128,75 @@ def set_z_index_by_z_coordinate(self):
...
 class Group(Mobject):
-    """Groups together multiple Mobjects."""
+    """A group of mobjects.
+
+    This can be used to group multiple :class:`~.Mobject` instances together
+    in order to scale, move, or perform other animations on them together.
+
+    To add :class:`~.Mobject`s to a :class:`~.Group`, you can either use the
+    :meth:`~.Group.add` method, or use the `+` and `+=` operators. Similarly, you
+    can subtract elements of a Group via :meth:`~.Group.remove` method, or
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:2138]: As this is currently written, Group doesn't actually override the remove method from the class Mobject, so we must either add the method to this class, or switch this to ~.Mobject.remove. I understand that this is mainly moving VGroup code to Group, but this should also probably be done in VGroup as well (I actually made those changes anyways so this is sorta my fault in retrospect)

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -2128,12 +2128,75 @@ def set_z_index_by_z_coordinate(self):
...
+        )
+
+    def __str__(self):
+        return (
+            f"{self.__class__.__name__} of {len(self.submobjects)} "
+            f"submobject{'s' if len(self.submobjects) > 0 else ''}"
+        )
+
+    def add(self, *mobjects):
+        """Checks if all passed elements are an instance of Mobject and then add them to submobjects
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:2165]:
```suggestion
        """Checks if all passed elements are an instance of :class:`Mobject` and then add them to :attr:`~Mobject.submobjects`
```

File: manim/mobject/mobject.py [hunk 4]
```diff
@@ -2128,12 +2128,75 @@ def set_z_index_by_z_coordinate(self):
...
+
+        Parameters
+        ----------
+        mobjects : :class:`~.Mobject`
+            List of Mobjects to add
+
+        Returns
+        -------
+        :class:`Group`
```
Comment (jsonvillanueva) [inline manim/mobject/mobject.py:2174]: One of my changes from VGroup that does have the `~.` -- should be fixed in VGroup as well.

**Review by jsonvillanueva — COMMENTED** [review]
More/less good to go, but the main issue from VGroup to Group is that mention of a method that Group (/VGroup) are supposed to override that doesn't actually exist.

File: manim/mobject/types/vectorized_mobject.py [hunk 5]
```diff
@@ -1292,7 +1292,7 @@ class VGroup(VMobject):
     """A group of vectorized mobjects.
 
     This can be used to group multiple :class:`~.VMobject` instances together
-    in order to scale, move, ... them together.
+    in order to scale, move, or perform other animations on them together.
```
Comment (cobordism) [inline manim/mobject/types/vectorized_mobject.py:1295]: ...but scale and move are not animations.

**Comment by WampyCakes** [comment]
Should I just close this in favor of revamping Manim's group system in general? cc @friedkeenan

**Comment by friedkeenan** [comment]
That may be wise, though I could foresee that PR generating a bit of discussion, so maybe it's best to get this simple one through first? Though we also have trouble getting simple PRs through lol

**Comment by WampyCakes** [comment]
I don't doubt we could get this through easily. Just not sure it's worth the updates this PR needs if it'll get blown away soonish. And we've gotten by fine without it thus far

**Comment by friedkeenan** [comment]
Yeah, that makes sense to me. Worst case scenario we can just reopen this PR or make a new one doing the same thing later down the line.

**Pull request closed without merging**